packages/react-scripts/scripts/lint.js

```diff
@@ -0,0 +1,28 @@
+// @remove-on-eject-begin
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+// @remove-on-eject-end
+'use strict';
+
+const CLIEngine = require('eslint').CLIEngine;
+
+const cli = new CLIEngine({
+  // @remove-on-eject-begin
+  configFile: require.resolve('eslint-config-react-app'),
+  // @remove-on-eject-end
+  fix: process.argv.slice(2).indexOf('--fix') >= 0,
+});
+const report = cli.executeOnFiles(['src/**/*.{js,jsx,mjs}']);
+const formatter = cli.getFormatter();
+
+// persist changed files when using --fix option
+CLIEngine.outputFixes(report);
+console.log(formatter(report.results));
+
+if (report.errorCount > 0) {
```

Comment:
Condition to trigger `exit(1)` could be different when a `--fix` flag was used.

Report object has a count of fixable errors and it is possible to count how many unfixed errors are left:

```js
{
  '...': '...',
  errorCount: 1,
  fixableErrorCount: 1,
}
```

My concern is that in `react-scripts lint --fix && git push` files will be changed by lint but they still need `git commit`.